src/main/scala/firrtl/stage/FirrtlOptions.scala

```diff
@@ -0,0 +1,93 @@
+import firrtl.annotations.Annotation
+import firrtl.ir.Circuit
+import logger.LogLevel
+
+/** Internal options used to control the FIRRTL compiler stage
+  *
+  * @param topName the name of the top module
+  * @param globalLogLevel the verbosity of logging (default: [[logger.LogLevel.None]])
+  * @param classLogLevels the individual verbosity of logging for specific classes
+  * @param logToFile if true, log to a file
+  * @param logClassNames indicates logging verbosity on a class-by-class basis
+  * @param programArgs explicit program arguments
+  * @param inputFileNameOverride input FIRRTL file, default: `targetDir/topName.fir`
+  * @param outputFileNameOverride output file, default: `targetDir/topName.SUFFIX` with `SUFFIX`
+  *                               as determined by the compiler
+  * @param outputAnnotationFileName the output annotation filename, default: `targetDir/topName.anno`
+  * @param compilerName which compiler to use (default: "verilog")
+  * @param infoModeName the policy for generating [[firrtl.ir.Info]] when processing FIRRTL (default: "append")
+  * @param customTransforms any custom [[Transform]] to run
+  * @param firrtlSource explicit input FIRRTL as a [[scala.String]]
+  * @param annotations a sequence of [[annotations.Annotation]] passed to the compiler
+  * @param emitOneFilePerModule enables one-file-per-module output in the [[Emitter]]
+  */
+final case class FirrtlOptions(
+  stageOptions:             StageOptions                = StageOptions(),
+  programArgs:              Seq[String]                 = Seq.empty,
+  inputFileNameOverride:    Option[String]              = None,
+  outputFileNameOverride:   Option[String]              = None,
+  outputAnnotationFileName: Option[String]              = None,
+  compilerName:             String                      = CompilerNameAnnotation().value,
+  infoModeName:             String                      = InfoModeAnnotation().value,
+  customTransforms:         Seq[Transform]              = List.empty,
+  firrtlSource:             Option[String]              = None,
+  annotations:              AnnotationSeq               = Seq.empty,
+  emitOneFilePerModule:     Boolean                     = false,
+  firrtlCircuit:            Option[Circuit]             = None) {
+
+  /** The top (main) in the FIRRTL Circuit
+    * @return the top, if a circuit exists, [[None]] otherwise
+    */
+  def topName: String = firrtlCircuit.map(_.main).getOrElse("a")
+
+  def outputFileName: String = outputFileNameOverride.getOrElse(topName)
+
+  /** Get the user-specified [[OutputConfig]]
+    *
+    * @param optionsManager this is needed to access build function and its common options
+    * @return the output configuration
+    */
+  def getOutputConfig(): OutputConfig =
+    if (emitOneFilePerModule) { OneFilePerModule(stageOptions.targetDir) }
+    else                      { SingleFile(stageOptions.getBuildFileName(outputFileName, Some(outputSuffix))) }
```

Comment:
outputSuffix is declared as requiring empty parens, this should change or the declaration should

Reply:
Apparently, side effect-free methods are supposed to omit parentheses: https://docs.scala-lang.org/style/method-invocation.html. I'll update everything in `FirrtlOptions` to follow that (as nothing in here has side effects).